pkg/credentials/credentials.go

```diff
@@ -234,6 +228,73 @@ func (f *fetcher) parseAWSCredentials() (map[string]string, error) {
 	return creds, nil
 }
 
+func openstackAuthOptionsToCreds(opts *gophercloud.AuthOptions, region string) (map[string]string, error) {
+	creds := make(map[string]string)
+	// We could implement some fallbacks here to support e.g. application credentials / domain ID
+	creds[OpenStackAuthURL] = opts.IdentityEndpoint
+	creds[OpenStackUserNameMC] = opts.Username
+	creds[OpenStackPassword] = opts.Password
+	creds[OpenStackDomainName] = opts.DomainName
+	creds[OpenStackRegionName] = region
+
+	if opts.TenantID != "" {
+		creds[OpenStackTenantID] = opts.TenantID
+	}
+
+	if v, ok := creds[OpenStackTenantID]; !ok || len(v) == 0 {
+		creds[OpenStackTenantName] = opts.TenantName
+	}
+
+	return creds, nil
+}
+
+func (f *fetcher) parseOpenStackCredentials() (map[string]string, error) {
+	// 1.) Credentials file loaded
+	if f.Source != nil {
+		return map[string]string{
+			OpenStackAuthURL:    f.F(OpenStackAuthURL),
+			OpenStackPassword:   f.F(OpenStackPassword),
+			OpenStackDomainName: f.F(OpenStackDomainName),
+			OpenStackRegionName: f.F(OpenStackRegionName),
+			OpenStackTenantID:   f.F(OpenStackTenantID),
+			OpenStackTenantName: f.F(OpenStackTenantName),
+			OpenStackUserNameMC: f.F(OpenStackUserName),
+		}, nil
+	}
+
+	// 2.) Check environment variables only use them if all required parameters are set
+	opts, err := openstack.AuthOptionsFromEnv()
```

Comment:
@johscheuer We just discussed this and we came with an idea to avoid using this function, but instead, write a function that would initialize `gophercloud.AuthOptions{}` manually. It's supposed to fetch and validate just the needed environment variables.

Does this idea sound good to you and can you modify the PR to implement it?